### PR TITLE
Implement the designs - bonus/episodes section

### PR DESCRIPTION
**Pull request by matthewthomsonnz**
This PR implements the meta detail bonus/episode section new designs. Notably there's a difficult shift in element arrangement/layout between mobile and tablet. I experimented with strategies and settled on this one as it appears to be the only option. (A black magic combination of float, clearfix, order, and margin-left).

Designs:
- [Mobile](https://app.zeplin.io/project/5fdaa165fbacf584a2ce4f8b/screen/6142ba83fd3fd250f50e6216)
- [Tablet](https://app.zeplin.io/project/5fdaa165fbacf584a2ce4f8b/screen/6142baeda13fc726053e405a)
- [Desktop](https://app.zeplin.io/project/5fdaa165fbacf584a2ce4f8b/screen/6142bb2eb8582a2b4c2c93ee)

Note:
- Mobile design shows collapsible on mobile. That will be implemented at a later date in relish, just need the layout there first.

[AB#6846](https://dev.azure.com/S72/fefacba9-96b6-4af2-a53d-050ed453e13a/_workitems/edit/6846)

### CODE REVIEW

**Review by brynyShift72 — COMMENTED** [review]
Seems the designs have the Bonus/Episodes title below the poster image, Currently looks as if its dependent on the size of the synopsis. eg
Zeplin:
image
This Branch:
image
Looks like the current implementation has it below the poster.

File: site/styles/_meta-detail.scss [hunk 1]
```diff
@@ -564,15 +462,141 @@
...
 }
 
-.meta-detail-bonus-content .synopsis {
-  font-size: 12px;
-  @include media-breakpoint-up(lg) {
-    font-size: 12px;
+.bonus-item,
+.episode-item {
+
+  @extend .clearfix;
```
Comment (PeculiarGoat) [inline site/styles/_meta-detail.scss:505]: No Bootstrap plz!

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
I feel like this is a good opportunity build something better from scratch rather than bending the existing mess to do something new.

Given that the bonus and episode are _almost_ identical, they can be extracted into a single shared jet.  And switching to grid should greatly simplify handling of the different layouts at different breakpoints.  And gap can take care of all of the padding and margins in a much simpler way.

**Comment by l0ud0gg** [comment]
a couple of nit picks, otherwise looks good (other than the others feedback);

1. thumbnail seems to have rounded corners on top but square on the bottom 
2. play button not centered

Screen Shot 2022-05-04 at 11 07 35 AM

**Review by l0ud0gg — APPROVED** [review]
If the others are happy, mine are nit picks, otherwise im happy :)

**Comment by matthewthomsonnz** [comment]
> Seems the designs have the Bonus/Episodes title below the poster image, Currently looks as if its dependent on the size of the synopsis. eg Zeplin: image This Branch: image Looks like the current implementation has it below the poster.

@brynyShift72 I could raise this with design but am pretty confident this is fine, even welcomed. The goal of the element switcher for example was to bring the episodes/bonuses up as high as possible, as some clients were complaining these sections are too low.

File: site/styles/_meta-detail.scss (hunk 1 above)
Comment (matthewthomsonnz) [reply site/styles/_meta-detail.scss]: - removed in favor of css grid

**Comment by matthewthomsonnz** [comment]
> a couple of nit picks, otherwise looks good (other than the others feedback);
> 
> 1. thumbnail seems to have rounded corners on top but square on the bottom
> 2. play button not centered
> 
> Screen Shot 2022-05-04 at 11 07 35 AM

Cheers Louis
- play button centered
- border radius bottom fixed

**Comment by matthewthomsonnz** [comment]
> I feel like this is a good opportunity build something better from scratch rather than bending the existing mess to do something new.
> 
> Given that the bonus and episode are _almost_ identical, they can be extracted into a single shared jet. And switching to grid should greatly simplify handling of the different layouts at different breakpoints. And gap can take care of all of the padding and margins in a much simpler way.

- Changed approach to grid
- Merged bonus_item/episode_item into single new block (called sub_item)

I concede grid is actually pretty good

**Review by smitsanghvi21 — COMMENTED** [review]
Not sue from where is this image coming from. Or is it showing some part of the landscape image?

 https://staging-store-kibble.shift72.com/admin/#/film/48/images

Screen Shot 2022-05-04 at 9 09 03 AM

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
Looking pretty good!

I'm wondering if we need to handle cases where clients upload oddly sized images:
image

Looks even stranger on mobile:
image

**Comment by matthewthomsonnz** [comment]
> Not sue from where is this image coming from. Or is it showing some part of the landscape image?
> 
> https://staging-store-kibble.shift72.com/admin/#/film/48/images
> 
> Screen Shot 2022-05-04 at 9 09 03 AM

Hey @smitsanghvi21 that ones unrelated. Pretty sure thats some old custom classification image work. I cleared that image once by copying the admin film save query into postman.

File: site/templates/items/sub_item.jet [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{{ block subItem(item) }}
+  <div class="poster-container">
```
[the remark this answers is not in the document]
Comment (matthewthomsonnz) [reply site/templates/items/sub_item.jet:2]: Do we need to though? It feels like complexity for no purpose

**Comment by matthewthomsonnz** [comment]
> Looking pretty good!
> 
> I'm wondering if we need to handle cases where clients upload oddly sized images: image
> 
> Looks even stranger on mobile: image

Cheers. Sure did look weird nice catch.

- Have showed Bex in person and added a fix

File: site/templates/items/sub_item.jet (hunk 2 above)
[the remark this answers is not in the document]
Comment (PeculiarGoat) [reply site/templates/items/sub_item.jet:2]: No probably not... it's just that jet changes are easier to review than CSS changes 😛

**Review by PeculiarGoat — COMMENTED** [review]
Good fix!

Do we care about things getting bent out of shape when there is too much content?
image

Probably a question for design team.

**Review by brynyShift72 — COMMENTED** [review]
@matthewthomsonnz Was there a take away/ answer yet from design on the Bonus/Episode placement in regards to its position relative to the poster?

**Comment by matthewthomsonnz** [comment]
> Good fix!
> 
> Do we care about things getting bent out of shape when there is too much content? 
> 
> Probably a question for design team.

Cheers. I showed Bex and she said that's fine as long as we maintain the same size margin under the heading

**Comment by matthewthomsonnz** [comment]
> @matthewthomsonnz Was there a take away/ answer yet from design on the Bonus/Episode placement in regards to its position relative to the poster?

Ah I forgot about that. Yes! There was. I'll have to add a wrapper to the main content with min height.

**Comment by matthewthomsonnz** [comment]
@brynyShift72 I've added a wrapper with min height. It works for the typical layout, but not if abnormal size poster or sponsor logo. We will have to remove this rule and redo whole detail page with CSS grid instead in a future PR.

**Comment by PeculiarGoat** [comment]
@matthewthomsonnz I don't see the point of that wrapper... I would prefer to not have it.  IMO the designs show that the Bonus Content/Episodes header should be relative to the See More button (always 42px below it) and not relative to the poster.

**Comment by matthewthomsonnz** [comment]
> @matthewthomsonnz I don't see the point of that wrapper... I would prefer to not have it. IMO the designs show that the Bonus Content/Episodes header should be relative to the See More button (always 42px below it) and not relative to the poster.

@PeculiarGoat Yeah I agree - reverted. Definitely appears relative in the designs, and it can't be treated as an afterthought we'll do it properly in another PR with grid

**Comment by stajs** [comment]
It looks like @l0ud0gg has deferred to the others, so if @smitsanghvi21 @brynyShift72 can review and give @PeculiarGoat a break!

**Review by PeculiarGoat — CHANGES_REQUESTED** [review]
Nearly there!

File: site/styles/_meta-detail.scss [hunk 3]
```diff
@@ -575,15 +474,130 @@ s72-element-switcher,
...
+    }
   }
 }
 
-.meta-detail-bonus-content .synopsis {
-  font-size: 12px;
-  @include media-breakpoint-up(lg) {
-    font-size: 12px;
+.bonus-item,
+.episode-item {
```
[the remark this answers is not in the document]
Comment (matthewthomsonnz) [reply site/styles/_meta-detail.scss]: - replaced bonus-item and episode item with sub-item
- moved .sub-item scss into a new file

File: site/styles/_meta-detail.scss [hunk 4]
```diff
@@ -575,15 +474,130 @@ s72-element-switcher,
...
+  h2 {
+    font-size: 18px;
+    margin-bottom: 15px;
+
+    @include media-breakpoint-up(sm) {
+      font-size: 22px;
+    }
+  }
+
+  .poster {
```
[the remark this answers is not in the document]
Comment (matthewthomsonnz) [reply site/styles/_meta-detail.scss]: - css moved

**Review by PeculiarGoat — APPROVED** [review]
Nice work @matthewthomsonnz!  It will be much easier to add/change things for bonus/episodes in the future now!

**Pull request merged**